### PR TITLE
430 platform flag

### PR DESCRIPTION
**Pull request by mtgoncalves1**
Issue https://gitlab.com/architect-io/architect-cli/-/issues/430.   

## Usage  
architect-local register . --architecture arch1 --architecture arch2

### CODE REVIEW

File: src/commands/register.ts [hunk 1]
```diff
@@ -113,8 +117,16 @@ export default class ComponentRegister extends Command {
           delete service.build.args;
         }
 
+        let platforms: string[];
+        try {
+          platforms = DockerBuildXUtils.convertToBuildxPlatforms(flags['platform']);
+        } catch (err: any) {
+          this.log('Failed to convert platform argument to docker buildx platforms. Please provide the correct platform(s).');
```
Comment (mueschm) [inline src/commands/register.ts:124]: No need for this message the error you throw is descript enough.

File: src/commands/register.ts [hunk 2]
```diff
@@ -36,6 +36,10 @@ export default class ComponentRegister extends Command {
       description: 'Tag to give to the new component',
       default: 'latest',
     }),
+    platform: Flags.string({
+      description: 'Platform(s) you are running on',
```
Comment (mueschm) [inline src/commands/register.ts:40]: Showcase how to do multiple platforms at the same time.

File: test/commands/register.test.ts [hunk 3]
```diff
@@ -65,6 +65,56 @@ describe('register', function () {
...
+
+        const config = fs.readFileSync('examples/fusionauth/config/kickstart.json');
+        expect(body.config.services.fusionauth.environment.KICKSTART_CONTENTS).to.eq(config.toString().trim());
+        return body;
+      })
+      .reply(200, {})
+    )
+    .stdout({ print })
+    .stderr({ print })
+    .command(['register', 'examples/fusionauth/architect.yml', '-t', '1.0.0', '--platform', 'amd64', '-a', 'examples'])
```
Comment (mueschm) [inline test/commands/register.test.ts:90]: Test multiple platforms.

**Review by mueschm — CHANGES_REQUESTED** [review]

File: src/commands/register.ts (hunk 2 above)
Comment (ryan-cahill) [reply src/commands/register.ts:40]: The property `multiple: true,` would probably be useful here

Comment (mtgoncalves1) [reply src/commands/register.ts:40]: Good idea. I added `multiple` to the flag.

File: src/commands/register.ts (hunk 1 above)
Comment (mtgoncalves1) [reply src/commands/register.ts]: Message was removed

**Review by ryan-cahill — CHANGES_REQUESTED** [review]

File: src/commands/register.ts [hunk 4]
```diff
@@ -113,8 +118,15 @@ export default class ComponentRegister extends Command {
           delete service.build.args;
         }
 
+        let buildx_platforms: string[];
+        try {
+          buildx_platforms = DockerBuildXUtils.convertToBuildxPlatforms(flags['architecture']);
+        } catch (err: any) {
+          this.error(err);
```
[the remark this answers is not in the document]
Comment (mtgoncalves1) [reply src/commands/register.ts]: Good question. I removed it.

**Review by ryan-cahill — APPROVED** [review]

**Pull request merged**